src/backend/x11/window_inner.rs

```diff
@@ -0,0 +1,238 @@
+/**
+A note for future contributors and maintainers:
+
+When editing this file, grab the nearest copy of the ICCCM. Following the ICCCM is paramount to
+X11 clients behaving properly and preventing scenarios such as windows not being resized in tiling
+window managers.
+
+Pay particular attention to "Section 4: Client to Window Manager Communication"
+
+A link to the ICCCM Section 4: https://tronche.com/gui/x/icccm/sec-4.html
+*/
+use crate::utils::{Logical, Size};
+
+use super::{Atoms, Window, WindowProperties, X11Error};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, Mutex, Weak,
+};
+use x11rb::{
+    connection::Connection,
+    protocol::{
+        xfixes::ConnectionExt,
+        xproto::{
+            self as x11, AtomEnum, ConnectionExt as _, CreateGCAux, CreateWindowAux, Depth, EventMask,
+            PropMode, Screen, UnmapNotifyEvent, WindowClass,
+        },
+    },
+    rust_connection::RustConnection,
+    wrapper::ConnectionExt as _,
+};
+
+impl From<Arc<WindowInner>> for Window {
+    fn from(inner: Arc<WindowInner>) -> Self {
+        Window(Arc::downgrade(&inner))
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct WindowInner {
+    // TODO: Consider future x11rb WindowWrapper
```

Comment:
This todo is outdated and can just be removed. `WindowWrapper` borrows the connection and thus cannot be used here... :-/